elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metric/ComputedMetric.java

```diff
@@ -0,0 +1,209 @@
+     * and {@code containingCls = FactTable.class}
+     * <pre>
+     * {@code
+     * public class FactTable {
+     *
+     *     {@literal @}MetricAggregation(sum.class)
+     *     Long sessions
+     *
+     *     {@literal @}MetricAggregation(sum.class)
+     *     Long timeSpent
+     *
+     *     {@literal @}MetricComputation(metricExpression = "timeSpent / sessions")
+     *     Float timeSpentPerSession
+     *
+     *     {@literal @}MetricComputation(metricExpression = "timeSpentPerSession / gameCount")
+     *     Float timeSpentPerGame
+     * }
+     * }
+     * </pre>
+     * One possible expanded JPQL metricExpression for {@code timeSpentPerSession} can be
+     * {@code com_oath_blah_FactTable.timeSpent / com_oath_blah_FactTable.sessions} and for {@code timeSpentPerGame} as
+     * {@code (com_oath_blah_FactTable.timeSpent/com_oath_blah_FactTable.sessions) / com_oath_blah_FactTable.gameCount}.
+     * {@link Schema} keeps such metricExpression in the form of {@code TABLE_PREFIX.timeSpent / TABLE_PREFIX.sessions}
+     * for {@code timeSpentPerSession} and
+     * {@code (TABLE_PREFIX.timeSpent/TABLE_PREFIX.sessions) / TABLE_PREFIX.gameCount} for
+     * {@code timeSpentPerGame}, where {@code TABLE_PREFIX} is specified by {@link #TABLE_PREFIX}.
+     * <p>
+     * <b>It is caller's responsibility to replace the {@link #TABLE_PREFIX} with the per-user specific value at
+     * runtime.</b>
+     */
+    public static final String TABLE_PREFIX = "<PREFIX>";
+
+    private static final long serialVersionUID = -5243438667754088900L;
+
+    @Getter
+    private final String name;
+
+    @Getter
+    private final String longName;
+
+    @Getter
+    private final String description;
+
+    @Getter
+    private final Class<?> dataType;
+
+    @Getter
+    private final List<Class<? extends Aggregation>> aggregations;
+
+    @Getter
+    private final String metricExpression;
+
+    public ComputedMetric(String metricField, Class<?> cls, EntityDictionary entityDictionary) {
+        Meta metaData = entityDictionary.getAttributeOrRelationAnnotation(cls, Meta.class, metricField);
+        Class<?> fieldType = entityDictionary.getType(cls, metricField);
+
+        this.name = metricField;
+        this.longName = metaData == null || metaData.longName().isEmpty() ? metricField : metaData.longName();
+        this.description = metaData == null || metaData.description().isEmpty() ? metricField : metaData.description();
+        this.dataType = fieldType;
+        this.aggregations = MetricUtils.extractAggregations(metricField, cls, entityDictionary);
+
+        if (aggregations.isEmpty()) {
+            String message = String.format("'%s' in '%s' has no aggregation.", metricField, cls.getCanonicalName());
+            log.error(message);
+            throw new IllegalStateException(message);
+        }
+
+        this.metricExpression = expandMetricExpression(metricField, cls, new HashSet<>(), entityDictionary);
```

Comment:
The metricExpression will be expanded at query time because it depends on the requested aggregation function.  We should not do this during construction.